tensorflow/python/layers/base.py

```diff
@@ -191,6 +191,18 @@ def add_weight(self, name, shape, dtype=None,
       RuntimeError: If called with partioned variable regularization and
         eager execution is enabled.
     """
+    
+    def _should_add_regularizer(variable, existing_variable_set):
+      result = True
+      if isinstance(variable, tf_variables.PartitionedVariable):
+        for var in variable:
+          if var in existing_variable_set:
+            result = False
+            break
+      else:
+        result = variable not in existing_variable_set
+      return result
```

Comment:
One more: Can you eliminate `result`? It's not needed, just `return False` in L200, and return what you are assigning to `result` in L203.